nym-node-status-agent/src/cli/mod.rs

```diff
@@ -0,0 +1,74 @@
+use crate::probe::GwProbe;
+use clap::{Parser, Subcommand};
+use nym_bin_common::bin_info;
+use std::sync::OnceLock;
+
+pub(crate) mod generate_keypair;
+pub(crate) mod run_probe;
+
+// Helper for passing LONG_VERSION to clap
+fn pretty_build_info_static() -> &'static str {
+    static PRETTY_BUILD_INFORMATION: OnceLock<String> = OnceLock::new();
+    PRETTY_BUILD_INFORMATION.get_or_init(|| bin_info!().pretty_print())
+}
+
+#[derive(Parser, Debug)]
+#[clap(author = "Nymtech", version, long_version = pretty_build_info_static(), about)]
+pub(crate) struct Args {
+    #[command(subcommand)]
+    pub(crate) command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+pub(crate) enum Command {
+    RunProbe {
+        #[arg(short, long, env = "NODE_STATUS_AGENT_SERVER_ADDRESS")]
+        server_address: String,
+
+        #[arg(short = 'p', long, env = "NODE_STATUS_AGENT_SERVER_PORT")]
+        server_port: u16,
+
+        /// base58-encoded private key
+        #[arg(long, env = "NODE_STATUS_AGENT_AUTH_KEY")]
+        ns_api_auth_key: String,
```

Comment:
it's not meant for passing via CLI: `clap` is just our convention because it handles loading from environment in a convenient way 

but for good measure, I removed the CLI option so now it's only loadable via env var